test/unit/settings_test.py

```diff
@@ -218,3 +224,34 @@ def test_immutability(self):
         with pytest.raises(TypeError):
             settings = EiaSettings()
             settings.eia860 = Eia860Settings()
+
+
+def test_partitions_with_json_normalize(pudl_etl_settings):
+    """Ensure the FERC1 and CEMS partitions normalize."""
+    datasets = pudl_etl_settings.get_datasets()
+
+    ferc_parts = json_normalize(datasets["ferc1"].partitions)
+    if list(ferc_parts.columns) != ["year"]:
+        raise AssertionError(
+            "FERC1 paritions should have year and state columns only, found:"
+            f"{ferc_parts}"
+        )
+
+    cems_parts = json_normalize(datasets["epacems"].partitions)
+    if list(cems_parts.columns) != ["year", "state"]:
+        raise AssertionError(
+            "CEMS paritions should have year and state columns only, found:"
+            f"{cems_parts}"
+        )
+
+
+def test_partitions_for_datasource_table(pudl_settings_fixture, pudl_etl_settings):
+    """Test whether or not we can make the datasource table."""
+    datasets = pudl_etl_settings.get_datasets()
+    ds = Datastore(local_cache_path=pudl_settings_fixture["data_dir"])
+    datasource = make_datasources_table(datasets, ds)
+    if datasource.empty and datasets.keys() != 0:
+        raise AssertionError(
+            "Datasource table is empty with the following datasets in the settings: "
+            f"{datasets.keys()}"
+        )
```

Comment:
even though the `make_datasources_table` function is in `etl.py`, it mostly exercises the settings so I stuck it in this module. but could certainly make a new `etl_test.py` module if that makes more sense.